### PR TITLE
Dev chat

### PR DESCRIPTION
**Pull request by JoannesJ**
I've added a page that runs the chat connected to a mock-server, so controlled messages can be send for testing.
Would have made the style update a bit easier FeelsPepoMan
@xDashh I've also snuck in the reduction of the buffer space at the bottom of chat you asked for.

### CODE REVIEW

**Comment by JoannesJ** [comment]
I suspected the loading of history was constantly re-trashing the document layout; take a look at this garbage.

chrome_2018-11-20_21-32-47

I managed to reduce it with a simple conditional on the ChatWindow locking/unlocking.
My naive performance measures went from 1800 ms to 350 ms 🎉 

Batching the DOM insertions should further reduce that.
Another day maybe.

File: assets/chat/js/chat.js [hunk 1]
```diff
@@ -540,7 +540,7 @@ class Chat {
 
         if(win === null)
             win = this.mainwindow
-        win.lock()
+        if(!this.backlogloading) win.lock();
```
Comment (KenanY) [inline assets/chat/js/chat.js]: Curious: if we end up _not_ getting this lock, do we want to ensure that the code between here and the unlock is _not_ executed? As in, should we use `win` at all without this lock?

**Comment by KenanY** [comment]
@xDashh Adding you since it seems like `assets/dev/dev-chat/history.json` has some chat logs (for testing purposes) and just wanna be sure that it is okay to add that here.

**Comment by JoannesJ** [comment]
I'll replace the logs with some `lorem ipsum`.
That's probably nicer regardless of whats actually in the logs.

**Comment by xDashh** [comment]
Ah thanks, yeah, I'd like to censor the names at least. Or if they are relevant for anything the content except the names. (Ideally both)

File: assets/chat/js/chat.js (hunk 1 above)
Comment (JoannesJ) [reply assets/chat/js/chat.js]: The ChatWindow adds the messages to the actual DOM, so in that sense we need to use it.
And seeing as `updateAndPin` is called after inserting the history, I don't think the locking matters.

**Comment by JoannesJ** [comment]
All names and messages are changed now!

**Comment by xDashh** [comment]
Merged in https://github.com/MemeLabs/chat-gui/commit/cb7b5ed68a020eeb0a834c2024ebe441498f886e. Thanks!

**Comment by xDashh** [comment]
Can confirm, reloading feels a lot faster! :tada:

**Comment by JoannesJ** [comment]
Nice, I thought you guys wouldn't notice with your 4GHz bullshit CPUs

**Comment by xDashh** [comment]
Frontends aren't CPU-bound, they are javascript-bound :^)

**Pull request closed without merging**